redfish-core/lib/ethernet.hpp

```diff
@@ -1313,84 +1343,54 @@ inline void handleDHCPPatch(const std::string& ifaceId,
         nextv6DHCPState = ipv6Active;
     }
 
-    bool nextDNS{};
-    if (v4dhcpParms.useDnsServers && v6dhcpParms.useDnsServers)
-    {
-        if (*v4dhcpParms.useDnsServers != *v6dhcpParms.useDnsServers)
-        {
-            messages::generalError(asyncResp->res);
-            return;
-        }
-        nextDNS = *v4dhcpParms.useDnsServers;
-    }
-    else if (v4dhcpParms.useDnsServers)
+    bool nextDNS{}, nextDNSv6{};
+    nextDNS = nextDNSv6 = ethData.dnsEnabled;
+    if (v4dhcpParms.useDnsServers)
```

Comment:
These removed changes in this method could have been retained, and only the setDHCPConfig() calls would have changed to send the separate set command to the dbus.
I think this code already had checked the v4 and v6 settings separately using v4dhcpParms and v6dhcpParms

Reply:
It was mandating v4dhcpParms and v6dhcpParms to have the same values by returning generalError when they're not. It was removed because different values for different network types are supported now.